### ISSUE TITLE
感觉SiameseDataset中有一点小问题，同时对损失函数有点不理解

### ISSUE THREAD

**Issue by diaoa1900** [bubbliiiing/Siamese-pytorch#20]
**您好，我在学习并使用您的代码中遇到了一些不懂的地方，希望可以被解惑，3q！**

**①感觉SiameseDataset中存在一点问题，这个different_c_index的值还是有可能和c相同的，将第64行的range(0, self.types - 1)修改为different_c就可以确保different_c_index取到与c不同的值**
utils.dataloader.py文件中
41    c               = random.randint(0, self.types - 1)
62    different_c         = list(range(self.types))
63    different_c.pop(c)
64    different_c_index   = np.random.choice(range(0, self.types - 1), 1)

**②SiameseDataset的__getitem__中index未使用，取出图片纯随机，会不会导致一些图片未被使用，一些图片重复使用呢？**

**③损失函数的计算是使用网络的输出和target进行计算，而精度使用sigmoid后的网络输出和target进行计算，为什么要这样做呢？有点儿困惑**
utils.utils_fit文件中
35    outputs = model_train(images)
36    output  = loss(outputs, targets)
53    equal       = torch.eq(torch.round(nn.Sigmoid()(outputs)), targets)

**Comment by bubbliiiing** [comment]
1、这个不会相同的
2、epoch多的情况下，统计上讲不会
3、这是pytorch计算方法的问题，需要加强一下基础

**Comment by diaoa1900** [comment]
> 1、这个不会相同的 2、epoch多的情况下，统计上讲不会 3、这是pytorch计算方法的问题，需要加强一下基础

谢谢答复！我刚才又去试了一下，发现确实是我弄错了(different_c_index是有可能和c相同，但current_c和c不会相同)

**Comment by bubbliiiing** [comment]
好滴
